src/yunohost/permission.py

```diff
@@ -0,0 +1,513 @@
+# -*- coding: utf-8 -*-
+
+""" License
+
+    Copyright (C) 2014 YUNOHOST.ORG
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program; if not, see http://www.gnu.org/licenses
+
+"""
+
+""" yunohost_permission.py
+
+    Manage permissions
+"""
+
+import grp
+import random
+
+from moulinette import m18n
+from moulinette.utils.log import getActionLogger
+from yunohost.utils.error import YunohostError
+from yunohost.user import user_list, user_group_list
+from yunohost.log import is_unit_operation
+
+logger = getActionLogger('yunohost.user')
+
+def user_permission_list(auth, app=None, permission=None, username=None, group=None):
+    """
+    List permission for specific application
+
+    Keyword argument:
+        app        -- an application OR sftp, xmpp (metronome), mail
+        permission -- name of the permission ("main" by default)
+        username   -- Username to get informations
+        group      -- Groupname to get informations
+
+    """
+
+    user_l = user_list(auth, ['uid'])['users']
+
+    permission_attrs = [
+        'cn',
+        'groupPermission',
+        'inheritPermission',
+        'URL',
+    ]
+
+    # Normally app is alway defined but it should be possible to set it
+    if app and not isinstance(app, list):
+        app = [app]
+    if permission and not isinstance(permission, list):
+        permission = [permission]
+    if not isinstance(username, list):
+        username = [username]
+    if not isinstance(group, list):
+        group = [group]
+
+    permissions = {}
+
+    result = auth.search('ou=permission,dc=yunohost,dc=org',
+                         '(objectclass=permissionYnh)', permission_attrs)
+
+    for res in result:
+        permission_name = res['cn'][0].split('.')[0]
+        try:
+            app_name = res['cn'][0].split('.')[1]
```

Comment:
Can't we do something more elegant like 

```python
permission_name, app_name = res['cn'][0].split('.')
```

?

Reply:
Done by fd76f554